frontend/src/views/main/profile/UserProfile.vue

```diff
@@ -97,373 +97,87 @@
-  }
-
-  @Watch("externalWorkerSelected")
-  @Watch("allMLWorkerSettings", {deep: true})
-  public mlWorkerSelected() {
-    if (this.allMLWorkerSettings.length) {
-      this.currentWorker = this.allMLWorkerSettings.find(value => value.isRemote === this.externalWorkerSelected) || null;
-      this.installedPackagesData = this.currentWorker !== null ?
-          Object.entries(this.currentWorker.installedPackages).map(([key, value]) => ({
-            name: key,
-            version: value
-          })) : [];
+    if (email.value && email.value !== currentProfile?.email) {
+      updatedProfile.email = email.value;
     }
-  }
-
-  get userProfile() {
-    return readUserProfile(this.$store);
-  }
-
-  public submit() {
-    (this.$refs.observer as any).validate().then(() => {
-      const currentProfile = readUserProfile(this.$store);
-      const updatedProfile: UpdateMeDTO = {};
-      if (this.displayName && this.displayName !== currentProfile?.displayName) {
-        updatedProfile.displayName = this.displayName;
-      }
-      if (this.email && this.email !== currentProfile?.email) {
-        updatedProfile.email = this.email;
-      }
-      if (Object.keys(updatedProfile).length > 0) {
-        dispatchUpdateUserProfile(this.$store, updatedProfile).then(() => {
-          this.editModeToggle = false;
-          this.resetFormData();
-        });
-      } else {
-        this.editModeToggle = false;
-        this.resetFormData();
-      }
-    });
-  }
-
-  public async generateToken() {
-    const loadingNotification = {content: 'Generating...', showProgress: true};
-    try {
-      commitAddNotification(this.$store, loadingNotification);
-      commitRemoveNotification(this.$store, loadingNotification);
-      this.apiAccessToken = await api.getApiAccessToken();
-    } catch (error) {
-      commitRemoveNotification(this.$store, loadingNotification);
-      commitAddNotification(this.$store, {content: 'Could not reach server', color: 'error'});
-    }
-  }
-
-  get giskardAddress() {
-    return window.location.hostname;
-  }
-
-  public async copyToken() {
-    await copyToClipboard(this.apiAccessToken?.id_token);
-    commitAddNotification(this.$store, {content: "Copied to clipboard", color: "success"});
-  }
-
-  public async saveGeneralSettings(settings: GeneralSettings) {
-    if (!settings.isAnalyticsEnabled) {
-      mixpanel.opt_out_tracking();
+    if (Object.keys(updatedProfile).length > 0) {
+      userStore.updateUserProfile(updatedProfile).then(() => {
+        editModeToggle.value = false;
+        resetFormData();
+      });
     } else {
-      mixpanel.opt_in_tracking();
+      editModeToggle.value = false;
+      resetFormData();
     }
-    this.appSettings!.generalSettings = await api.saveGeneralSettings(settings);
-  }
+  });
+}
 
-  private epochToDate(epoch: number) {
-    return moment.unix(epoch).format('DD/MM/YYYY HH:mm:ss');
+async function generateToken() {
```

Comment:
we lost the possibility to generate API tokens in the free version since UserProfile vue isn't available